src/onegov/feriennet/models/activity.py

```diff
@@ -21,9 +24,28 @@ class VacationActivity(Activity, CoordinatesExtension, SearchableContent):
         'title': {'type': 'localized'},
         'lead': {'type': 'localized'},
         'text': {'type': 'localized_html'},
+        # FIXME: properties cannot be used for ts tsvector
         'organiser': {'type': 'text'}
     }
 
+    # column for full text search index
+    fts_idx = Column(TSVECTOR)
+
+    @property
+    def search_score(self):
+        return 2
+
+    @staticmethod
+    def psql_tsvector_string(model):
+        """
+        index is built on column title as well as the json
+        fields lead and text in meta resp. content column
+        """
+        s = Searchable.create_tsvector_string('title')
+        s += " || ' ' || coalesce(((meta ->> 'lead')), '')"
+        s += " || ' ' || coalesce(((content ->> 'text')), '')"
+        return s
+
```

Comment:
This should already be handled by the Mixin, no?

To make `organiser` work turn it into a `hybrid_property`. Writing the equivalent in SQL isn't trivial, but it should be possible with some combination of arrays, json operators and coalesce.

You could probably also get away with implementing something that isn't fully equivalent (i.e. contains some elements with empty strings or returns a concatenated string where all the keywords are separated by at least one space)

Reply:
Correct, removed